receiver/snmpreceiver/scraper.go

```diff
@@ -186,15 +217,22 @@ func (s *snmpScraper) indexedDataToMetric(
 	}
 
 	// Get resource attributes
-	resourceAttributes, err := getResourceAttributes(configHelper, data.columnOID, indexString, columnOIDIndexedResourceAttributeValues)
+	resourceAttributes, err := getResourceAttributes(configHelper, data.columnOID, indexString, columnOIDIndexedResourceAttributeValues, columnOIDScalarResourceAttributeValues)
 	if err != nil {
 		return fmt.Errorf(errMsgOIDResourceAttributeEmptyValue, metricName, err)
 	}
 
 	// Create a resource key using all of the relevant resource attribute names along
 	// with the row index of the SNMP data
 	resourceAttributeNames := configHelper.getResourceAttributeNames(data.columnOID)
-	resourceKey := getResourceKey(resourceAttributeNames, indexString)
+
+	var resourceKey string
+	// If we only have scalar resource attributes, we don't need multiple resources
```

Comment:
any reason why we couldn't just have all the scalar attributes as part of the key in the case of only scalar resource attributes?

Reply:
Sure, makes sense